tests/api/test_errors.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 The Matrix.org Foundation C.I.C.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from synapse.api.errors import LimitExceededError
+
+from tests import unittest
+
+
+class ErrorsTestCase(unittest.TestCase):
+    def test_limit_exceeded_header(self) -> None:
+        err = LimitExceededError(retry_after_ms=100)
+        self.assertEqual(err.error_dict(None).get("retry_after_ms"), 100)
+        assert err.headers
```

Comment:
I assume all of these are `assert err.headers is not None`?

Reply:
Sure, do we need to be that specific?

Reply:
Mostly just trying to understand what we're asserting. Not a big deal, I don't think. 👍

Reply:
I've done it anyway, since I was refactoring for the optional headers request below.